DQM/HcalTasks/plugins/QIE10Task.cc

```diff
@@ -0,0 +1,160 @@
+
+#include "DQM/HcalTasks/interface/QIE10Task.h"
+
+using namespace hcaldqm;
+QIE10Task::QIE10Task(edm::ParameterSet const& ps):
+	DQTask(ps)
+{
+	
+	//	tags
+	_tagQIE10 = ps.getUntrackedParameter<edm::InputTag>("tagQIE10",
+		edm::InputTag("hcalDigis"));
+	_tokQIE10 = consumes<QIE10DigiCollection>(_tagQIE10);
+
+	//	cuts
+	_cut = ps.getUntrackedParameter<double>("cut", 50.0);
+	_ped = ps.getUntrackedParameter<int>("ped", 4);
+}
+/* virtual */ void QIE10Task::bookHistograms(DQMStore::IBooker &ib,
+	edm::Run const& r, edm::EventSetup const& es)
+{
+	if (_ptype==fLocal)
+		if (r.runAuxiliary().run()==1)
+			return;
+
+	DQTask::bookHistograms(ib, r, es);
+
+	//	GET WHAT YOU NEED
+	edm::ESHandle<HcalDbService> dbs;
+	es.get<HcalDbRecord>().get(dbs);
+	_emap = dbs->getHcalMapping();
+	std::vector<uint32_t> vhashC36;
+	vhashC36.push_back(HcalElectronicsId(36, 3,
+		FIBER_uTCA_MIN1, FIBERCH_MIN, false).rawId());
+	_filter_C36.initialize(filter::fPreserver, hashfunctions::fCrateSlot,
+		vhashC36);
+
+	//	INITIALIZE what you need
+	_cShapeCut_EChannel.initialize(_name,
+		"ShapeCut", hashfunctions::fEChannel,
+		new quantity::ValueQuantity(quantity::fTiming_TS),
```

Comment:
These are properly cleaned and destroyed.
But I would advise to move those _qx members to unique_ptr in ContainerSingle2D.